apps/events/models/Attendance.py

```diff
@@ -23,6 +25,79 @@
 from .Extras import Extras
 
 
+@unique
+class StatusCode(IntEnum):
+    """
+    These codes follow the HTTP standard in terms of overlying scheme.
+    2XX = successful
+    4XX = client error (user related)
+    5XX = server error (event related)
+    These codes are meant as a debugging tool only. The eligibility checking is quite
+    extensive, and tracking where it's going wrong is much needed.
+    """
+
+    # Event has no rulebundles
+    SUCCESS_MEMBER = 200
+    SUCCESS_GUEST_SIGNUP = 201
+
+    SUCCESS_FIELD_OF_STUDY = 210
+    SUCCESS_GRADE = 211
+    SUCCESS_USER_GROUP = 212
+    SUCCESS_UNKNOWN = 213
+
+    NOT_SATISFIED = 500
+    SIGNUP_NOT_OPENED_YET = 501
+    SIGNUP_CLOSED = 502
+    NO_SEATS_LEFT = 503
+
+    NOT_A_MEMBER = 400
+    DELAYED_SIGNUP_MARKS = 401
+    DELAYED_ACCESS = 402
+    ACCESS_DENIED = 403
+    ALREADY_SIGNED_UP = 404
+    SUSPENDED = 405
```

Comment:
This is the only new code I believe, since previously `DELAYED_ACCESS = 402` was wrongly (?) used